### ISSUE TITLE
Option to make the repository list always visible

### ISSUE THREAD

**Issue by stijnherreman** [ForkIssues/TrackerWin#344]
I'm a SourceTree refugee, happy that (in the beta version) they finally added a dark theme and re-added a repository list sidebar, but still unhappy with overall performance. Fork's performance is a delight!

One thing I'm missing is an always-visible repository list, because the project I currently work on has 15 repositories and I often interact with multiple of them at the same time.

This could be a separate sidebar, or it could use the available space in the current sidebar:
![free space below the branches/remotes/tags/stashes/submodules of the open repository](https://user-images.githubusercontent.com/1618054/60190463-f6540900-9832-11e9-8fb8-badd2cc40613.png)

I'd be happy with a simple list of repositories, just like the list of the Repository Manager as seen when you open a new tab. A nice-to-have would be some info like the current branch name, number of incoming/outgoing commits, etc. but keeping that information up-to-date could have a noticeable impact on performance.

Another option is to add a vertical mode to the tab bar, since tabs can be ordered and open tabs are remembered between sessions.

What do you think?

**Comment by jerone** [comment]
There is already an Repository Manager in an new tab (<kbd>Ctrl</kbd>+<kbd>T</kbd>) and there is the "Open Quickly" feature (<kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>O</kbd>). I see no reason to clutter the left over space. I always have my branches and remotes open, so much room is not left anyways.

**Comment by stijnherreman** [comment]
@jerone I completely understand that a lot of people have no use for this and don't want to clutter their interface, that's why the title of this issue is "**Option** to make [...]".

**Comment by Otiel** [comment]
👎 as well for adding such a list to the bottom of the existing sidebar, as many people already have several branches, remotes and tags, which make the left-over space small if not non-existent. Moreover, it doesn't feel right to mix a list of repositories and "properties" of a single repository on the same sidebar.

However, I could see the need for vertical tabs when you have 15+ repositories opened, all the more if this feature would come to life:

>A nice-to-have would be some info like the current branch name, number of incoming/outgoing commits, etc.

**Comment by jerone** [comment]
> However, I could see the need for vertical tabs when you have 15+ repositories opened, all the more if this feature would come to life:
> 
> > A nice-to-have would be some info like the current branch name, number of incoming/outgoing commits, etc.

I thought there was an issue for this, but I can't find it. 

:+1: on all those things quoted; vertical tabs, current branch name in tab, number of incoming/outgoing commits in tab.

**Comment by Otiel** [comment]
> I thought there was an issue for this, but I can't find it.

@jerone #268?

**Comment by DanPristupov** [comment]
@jerone @Otiel  There's a problem with showing status for each repo in repo manager. We have this feature implemented in a feature branch, but I think it's not ready for production yet. 

Generally speaking, `git status` makes a diff between working directory and the repository index. This can be a **very** expensive operation for large repos. So, having some 2GB repos in the default source directory can screw up the overall performance and also drain battery on laptops if you keep the repo manager open.

For example, I keep the Chrome repo for testing purposes, which is 22GB and `git status` takes at least 10 seconds and full CPU loading.

I'm very careful introducing new features which can affect the performance.

@stijnherreman Fork has tabs and I think such sidebar will not fit into our design. Tabbed layout, repository manager, the Open Quickly window (Ctrl+Shift+O) and list of repos on the sidebar will really be a mess together.

I think the Open Quickly dialog is the closest thing you should try to use.

**Comment by jerone** [comment]
@DanPristupov Makes sense, you don't want to affect Fork's performance. 

Just thinking out loud; what if you introduce an concept of "**live and inactive tabs**". Where "_inactive_" tabs are just like right now, they only check for updates when they are open/focused. "_Live_" tabs check and show the status in all situations. I see this as an **opt-in** setting by the user (maybe with some warning), because they know when a repo is small or their connection is fast. And visual different (tab color maybe) to make note they are "live".

**Comment by DanPristupov** [comment]
Every option decrease the number of users by 10, unfortunately.

There are many possible heuristics. One of them could be the last time usage. If last time a repo has been updated was 1 month ago, then we can not update it so often, only once a day may be.
